### PR TITLE
Add ArchReports and IssueEntry object

### PR DESCRIPTION
**Pull request by asdil12**
- Obsolete Report object
- Fix soft failed by using new `result_softfailed` class string

### CODE REVIEW

**Comment by okurz** [comment]
I merged "Fix soft failed: Use new 'result_softfailed' str"

The rest is a mixture of refactoring and adding more information to the reports which is what the test verbosity is for. The terse report should be "terse" and should not include the URLs, it's not anymore

File: openqa_review/openqa_review.py [hunk 1]
```diff
@@ -505,76 +465,133 @@ def __str__(self):
...
+        self.root_url = root_url
+        if len(self.failures) == 0:
+            raise ValueError
+
+    def _url(self, v):
+        """Absolute url e.g. for test references."""
+        return absolute_url(self.root_url, v)
+
+    def _format_failure_modules(self, failedmodules):
+        return ', '.join(map(lambda m: m['name'], failedmodules))
```
Comment (okurz) [inline openqa_review/openqa_review.py]: use list comprehension or generator expressions, see
http://leadsift.com/loop-map-list-comprehension/
https://wiki.python.org/moin/PythonSpeed/PerformanceTips#Loops

File: openqa_review/openqa_review.py [hunk 2]
```diff
@@ -505,76 +465,133 @@ def __str__(self):
...
-            issues['new']['todo'] = simple_joined_issues(results_by_bugref, 'NEW_ISSUE')
-            issues['existing']['todo'] = simple_joined_issues(results_by_bugref, 'STILL_FAILING')
-        else:
-            report = Report(self.root_url, verbose_test)
-            issues['new']['todo'] = '\n'.join('* %s' % report.new_issue(k, v) for k, v in iteritems(results) if v['state'] == 'NEW_ISSUE')
-            issues['existing']['todo'] = '* ' + ', '.join(k for k, v in iteritems(results) if v['state'] == 'STILL_FAILING')
+        self.issues['new']['todo'].extend(IssueEntry.for_each(self.args, self.root_url,
+                                                              filter(lambda r: r['state'] == 'NEW_ISSUE', results_by_bugref.get('TODO', []))))
+        self.issues['existing']['todo'].extend(IssueEntry.for_each(self.args, self.root_url,
+                                                                   filter(lambda r: r['state'] == 'STILL_FAILING', results_by_bugref.get('TODO', []))))
```
Comment (okurz) [inline openqa_review/openqa_review.py]: also here, see http://www.u.arizona.edu/~erdmann/mse350/topics/list_comprehensions.html#filtered-list-comprehensions

File: openqa_review/openqa_review.py [hunk 3]
```diff
@@ -505,76 +465,133 @@ def __str__(self):
...
+            new_soft_fails = filter(lambda r: r['state'] == 'NEW_SOFT_ISSUE', results.values())
+            try:
+                self.issues['new']['product'].append(IssueEntry(self.args, self.root_url, new_soft_fails, soft=True))
+            except ValueError:
+                pass
+            existing_soft_fails = filter(lambda r: r['state'] == 'STILL_SOFT_FAILING', results.values())
+            try:
+                self.issues['existing']['product'].append(IssueEntry(self.args, self.root_url, existing_soft_fails, soft=True))
+            except ValueError:
+                pass
```
Comment (okurz) [inline openqa_review/openqa_review.py]: does not look nice

File: tests/report25_T.md [hunk 4]
```diff
@@ -14,14 +14,12 @@
 
 **New Product bugs:**
 
-
-* soft fails: upgrade_offline_13.1_allpatterns
+* soft fails: [upgrade_offline_13.1_allpatterns](https://openqa.opensuse.org/tests/169809 "Failed modules: ")
```
Comment (okurz) [inline tests/report25_T.md]: That shouldn't include any "Failed modules" if there aren't any

File: tests/report25_TT.md [hunk 5]
```diff
@@ -14,14 +14,12 @@
...
-
-* soft fails: upgrade_offline_13.1_allpatterns
+* soft fails: [upgrade_offline_13.1_allpatterns](https://openqa.opensuse.org/tests/169809 "Failed modules: ") [(ref)](https://openqa.opensuse.org/tests/169614 "Previous test")
 
 
 **Existing Product bugs:**
 
-
-* soft fails: autoupgrade_13.1, cryptlvm, cryptlvm_minimal_x, textmode+awesome
+* soft fails: [autoupgrade_13.1](https://openqa.opensuse.org/tests/169791 "Failed modules: ") [(ref)](https://openqa.opensuse.org/tests/169596 "Previous test"), [cryptlvm](https://openqa.opensuse.org/tests/169796 "Failed modules: zypper_in") [(ref)](https://openqa.opensuse.org/tests/169601 "Previous test"), [cryptlvm_minimal_x](https://openqa.opensuse.org/tests/169797 "Failed modules: ") [(ref)](https://openqa.opensuse.org/tests/169602 "Previous test"), [textmode+awesome](https://openqa.opensuse.org/tests/169828 "Failed modules: ") [(ref)](https://openqa.opensuse.org/tests/169633 "Previous test")
```
Comment (okurz) [inline tests/report25_TT.md]: "soft fails" are less important, therefore by default they should have less verbosity

File: tests/report25_TTT.md [hunk 6]
```diff
@@ -1,7 +1,7 @@
 # 0
 
 
-**Date:** $now
+**Date:** 2016-09-15 - 16:31
```
Comment (okurz) [inline tests/report25_TTT.md]: better keep $now. Maybe you can also replace the timestamp in the other reports but in another commit

File: tests/report25_terse.md [hunk 7]
```diff
@@ -41,9 +39,15 @@
 * gnome@uefi
 * textmode
 
+
```
Comment (okurz) [inline tests/report25_terse.md]: If you need this line please add in another commit

File: tests/report25_terse.md [hunk 8]
```diff
@@ -41,9 +39,15 @@
...
 ***existing issues***
 
-* gcc5, gcc5+allpatterns, toolchain_zypper, upgrade_offline_13.1+gcc5_64bit, upgrade_offline_13.2_gcc5
+* gcc5
+* gcc5+allpatterns
+* toolchain_zypper
+* upgrade_offline_13.1+gcc5_64bit
+* upgrade_offline_13.2_gcc5
+
```
Comment (okurz) [inline tests/report25_terse.md]: should still be all in one line

File: tests/tags_labels/report25_bugrefs.md [hunk 9]
```diff
@@ -11,26 +11,26 @@
...
 
 
 **Existing openQA-issues:**
 
-* textmode -> [poo#11948](https://progress.opensuse.org/issues/11948)
-* allpatterns@i586--l2 -> [poo#9772](https://progress.opensuse.org/issues/9772)
+* [textmode](https://openqa.opensuse.org/tests/384342 "Failed modules: dns_srv") [(ref)](https://openqa.opensuse.org/tests/382507 "Previous test") -> [poo#11948](https://progress.opensuse.org/issues/11948)
+* [allpatterns@i586--l2](https://openqa.opensuse.org/tests/384335 "Failed modules: ") [(ref)](https://openqa.opensuse.org/tests/382500 "Previous test") -> [poo#9772](https://progress.opensuse.org/issues/9772)
```
Comment (okurz) [inline tests/tags_labels/report25_bugrefs.md]: better keep this report for bugrefs with lowest test verbosity and add support for test verbosity together with bugrefs

File: tests/tags_labels/report25_bugrefs_query_issues.md [hunk 10]
```diff
@@ -11,26 +11,26 @@
...
 
 
 **Existing openQA-issues:**
 
-* textmode -> [poo#11948](https://progress.opensuse.org/issues/11948 "can not boot pre-installed image") (Ticket status: New, assignee: user1)
-* allpatterns@i586--l2 -> [poo#9772](https://progress.opensuse.org/issues/9772 "Nothing works") (Ticket status: New, assignee: None)
+* [textmode](https://openqa.opensuse.org/tests/384342 "Failed modules: dns_srv") [(ref)](https://openqa.opensuse.org/tests/382507 "Previous test") -> [poo#11948](https://progress.opensuse.org/issues/11948 "can not boot pre-installed image") (Ticket status: New, assignee: user1)
+* [allpatterns@i586--l2](https://openqa.opensuse.org/tests/384335 "Failed modules: ") [(ref)](https://openqa.opensuse.org/tests/382500 "Previous test") -> [poo#9772](https://progress.opensuse.org/issues/9772 "Nothing works") (Ticket status: New, assignee: None)
```
Comment (okurz) [inline tests/tags_labels/report25_bugrefs_query_issues.md]: Adding test verbosity together with querying the issue status should be separated. Let's do one thing after another

File: openqa_review/openqa_review.py [hunk 11]
```diff
@@ -505,76 +465,133 @@ def __str__(self):
...
+    def _format_failure(self, f):
+        """Yield a report entry for one new issue based on verbosity."""
+        if self.args.verbose_test >= 3 and 'prev' in f:
+            return '[%s](%s "Failed modules: %s") [(ref)](%s "Previous test")' % (
+                f['name'], self._url(f), self._format_failure_modules(f['failedmodules']), self._url(f['prev'])
+            )
+        elif self.args.verbose_test >= 2:
+            return '[%s](%s "Failed modules: %s")' % (f['name'], self._url(f), self._format_failure_modules(f['failedmodules']))
+        else:
+            return '%s' % f['name']
```
Comment (okurz) [inline openqa_review/openqa_review.py]: Please use a dispatch table instead of if/elif/else

File: openqa_review/openqa_review.py (hunk 1 above)
Comment (asdil12) [inline openqa_review/openqa_review.py]: done

File: openqa_review/openqa_review.py (hunk 2 above)
Comment (asdil12) [inline openqa_review/openqa_review.py]: done

File: openqa_review/openqa_review.py (hunk 3 above)
Comment (asdil12) [inline openqa_review/openqa_review.py]: done

File: tests/report25_T.md (hunk 4 above)
Comment (asdil12) [inline tests/report25_T.md]: done

File: tests/report25_TT.md (hunk 5 above)
Comment (asdil12) [inline tests/report25_TT.md]: done (fixed tests setting -TTT by default)

File: tests/tags_labels/report25_bugrefs.md (hunk 9 above)
Comment (asdil12) [inline tests/tags_labels/report25_bugrefs.md]: done (fixed tests setting -TTT by default)

**Pull request merged**